File: project/swig/ruby/logging.rb
#!/usr/bin/env ruby

require 'librets'

# Here is where we start doing our RETS stuff
session =
  Librets::RetsSession.new("http://demo.crt.realtors.org:6103/rets/login")

log_file = ARGV[0]
session.SetHttpLogName(log_file)

if !session.login("Joe", "Schmoe")
  puts "Invalid login"
  exit 1
end

logout = session.logout

# If we are using logging at ALL, we must call cleanup on session.
# This will free any native resources that point to a ruby based
# logging object.
session.cleanup

puts "See the logging in file #{log_file}" unless log_file.nil?

File: project/swig/python/logging.py
#!/usr/bin/env python
from __future__ import print_function
from argparse import ArgumentParser
import sys

import librets

try:
    p = ArgumentParser()
    p.add_argument('log_file')
    args = p.parse_args()

    session = librets.RetsSession("http://www.dis.com:6103/rets/login")
    session.SetHttpLogName(args.log_file)

    if not session.Login("Joe", "Schmoe"):
        sys.exit("Invalid login")

    logout = session.Logout()
    session.Cleanup()

except librets.RetsException as e:
    print("Caught:", e.GetMessage())
